pkg/datapath/loader/netlink.go

```diff
@@ -329,3 +332,58 @@ func SetupBaseDevice(mtu int) (netlink.Link, netlink.Link, error) {
 
 	return linkHost, linkNet, nil
 }
+
+// reloadIPSecOnLinkChanges subscribes to link changes to detect newly added devices
+// and reinitializes IPsec on changes. Only in effect for ENI mode in which we expect
+// new devices at runtime.
+func (l *Loader) reloadIPSecOnLinkChanges() {
+	// settleDuration is the amount of time to wait for further link updates
+	// before proceeding with reinitialization. This avoids back-to-back
+	// reinitialization when multiple link changes are made at once.
+	const settleDuration = 1 * time.Second
```

Comment:
The way I read this comment initially, I assumed this was done for every time we observe a burst of link changes. But is seems we only coalesce events for the initial round. Is that intentional?

Reply:
If that's the intent, then the code is currently broken. `timer, stop := inctimer.New()` only fires once and it's never reset

Reply:
looking at `After` it seems like it cancels old timer and creates a new one. Did I misunderstand the API?

Reply:
Though what was a bit unintended was resetting the timer after every update. I've now refactored the code a bit. PTAL.

Reply:
Oh, sorry, you are of course right. I forgot that `inctimer` is started when we call `After`, not when we call `inctimer.New()` :facepalm: 

Sorry for the noise.